esp-wifi/src/lib.rs

```diff
@@ -364,9 +364,9 @@ impl private::Sealed for Trng<'_> {}
 /// .unwrap();
 /// # }
 /// ```
-pub fn init<'d, T: EspWifiTimerSource>(
+pub fn init<'d, T: EspWifiTimerSource, R: EspWifiRngSource>(
     timer: impl Peripheral<P = T> + 'd,
-    _rng: impl EspWifiRngSource,
+    _rng: impl Peripheral<P = R> + 'd,
```

Comment:
```suggestion
    _rng: impl Peripheral<P = RNG> + 'd,
```

You might need to import `esp_hal::peripherals::RNG`.

And with that, you can remove the EspWifiRngSource trait I think

Reply:
I've done that, please have another look.